vendor/github.com/samuel/go-zookeeper/zk/dnshostprovider.go

```diff
@@ -0,0 +1,88 @@
+package zk
+
+import (
+	"fmt"
+	"net"
+	"sync"
+)
+
+// DNSHostProvider is the default HostProvider. It currently matches
+// the Java StaticHostProvider, resolving hosts from DNS once during
+// the call to Init.  It could be easily extended to re-query DNS
+// periodically or if there is trouble connecting.
+type DNSHostProvider struct {
```

Comment:
Well, not only that, but it looks like the Zookeeper client has a little bit of internal state. If we go into a disconnect state, does it make sense to try to completely rebuild the client after some interval to get rid of any potential bad state?